src/dahua.ts

```diff
@@ -28,51 +27,57 @@ class DahuaEvents {
     public RECONNECTING_EVENT_NAME: string = 'reconnecting'
 
     private host:                   string
+    private user:                   string
+    private pass:                   string
+    private requestConfig:          AxiosRequestConfig
 
     constructor(host: string, user: string, pass: string, useHttp: boolean) {
         this.host = host
-        const auth: AxiosBasicCredentials = {
-            username: user,
-            password: pass
-        }
+        this.user = user
+        this.pass = pass
 

```

Comment:
Would prefer to keep this than add the user/pass class variables. This object will do the basic auth stuff for us (base 64 encoding and such) and then gets overrode if digest auth comes to play.